ome_zarr/napari.py

```diff
@@ -0,0 +1,82 @@
+"""This module is a napari plugin.
+
+It implements the ``napari_get_reader`` hook specification, (to create a reader plugin).
+"""
+
+
+import logging
+import warnings
+from typing import Any, Callable, Dict, Iterator, List, Optional
+
+from .data import CHANNEL_DIMENSION
+from .io import parse_url
+from .reader import Label, Layer, Reader
+from .types import LayerData, PathLike, ReaderFunction
+
+try:
+    from napari_plugin_engine import napari_hook_implementation
+except ImportError:
+
+    def napari_hook_implementation(
+        func: Callable, *args: Any, **kwargs: Any
+    ) -> Callable:
+        return func
+
+
+LOGGER = logging.getLogger("ome_zarr.napari")
+
+
+@napari_hook_implementation
+def napari_get_reader(path: PathLike) -> Optional[ReaderFunction]:
+    """Returns a reader for supported paths that include IDR ID.
+
+    - URL of the form: https://s3.embassy.ebi.ac.uk/idr/zarr/v0.1/ID.zarr/
+    """
+    if isinstance(path, list):
+        if len(path) > 1:
+            warnings.warn("more than one path is not currently supported")
+        path = path[0]
+    zarr = parse_url(path)
+    if zarr:
+        reader = Reader(zarr)
+        return transform(reader())
+    # Ignoring this path
+    return None
+
+
+def transform(layers: Iterator[Layer]) -> Optional[ReaderFunction]:
+    def f(*args: Any, **kwargs: Any) -> List[LayerData]:
+        results: List[LayerData] = list()
+
+        for layer in layers:
+            data: List[Any] = layer.data
+            metadata: Dict[str, Any] = layer.metadata
+            if data is None or len(data) < 1:
+                LOGGER.debug(f"skipping non-data {layer}")
+            else:
+                LOGGER.debug(f"transforming {layer}")
+                shape = data[0].shape
```

Comment:
The `shape` I have here is `(3, 1, 1024, 1344)` so the `channel_axis` is not added below.

Reply:
Pushed a fix.